File: apps/web/lib/audit-logs.ts
export type AuditLogEntry = {
  id: string;
  action: string;
  actor: string;
  target: string;
  createdAt: string;
  metadata: Record<string, string>;
};

export const demoAuditLogs: AuditLogEntry[] = [
  {
    id: "audit_document_uploaded",
    action: "document.upload_registered",
    actor: "demo.admin@example.com",
    target: "doc_upload_synthetic_vendor_policy_md",
    createdAt: "2026-01-15T12:00:00Z",
    metadata: {
      mode: "synthetic_demo_upload",
      status: "accepted_for_processing"
    }
  },
  {
    id: "audit_document_processed",
    action: "document.processed",
    actor: "system",
    target: "doc_vendor_intake",
    createdAt: "2026-01-15T12:01:00Z",
    metadata: {
      chunkCount: "3",
      nextStep: "ready_for_rag_indexing"
    }
  },
  {
    id: "audit_rag_query_submitted",
    action: "rag.query_submitted",
    actor: "demo.admin@example.com",
    target: "airun_what_risks_a",
    createdAt: "2026-01-15T12:30:00Z",
    metadata: {
      provider: "mock",
      sourceCount: "2"
    }
  },
  {
    id: "audit_ai_run_created",
    action: "ai_run.created",
    actor: "demo.admin@example.com",
    target: "airun_vendor_risk_summary",
    createdAt: "2026-01-15T12:10:00Z",
    metadata: {
      workflow: "document_intake_review",
      status: "needs_review"
    }
  },
  {
    id: "audit_review_created",
    action: "review.created",
    actor: "system",
    target: "review_vendor_risk_summary",
    createdAt: "2026-01-15T12:12:00Z",
    metadata: {
      priority: "high",
      sourceDocumentId: "doc_vendor_intake"
    }
  },
  {
    id: "audit_review_approved",
    action: "review.approved",
    actor: "demo.reviewer@example.com",
    target: "review_vendor_risk_summary",
    createdAt: "2026-01-15T12:42:00Z",
    metadata: {
      auditability: "sources_verified",
      decision: "approved"
    }
  }
];

export function getAuditLogs(): AuditLogEntry[] {
  return demoAuditLogs;
}

File: apps/api/app/services/audit_logs.py
from datetime import UTC, datetime

from app.models.audit_logs import AuditLogEntry


def get_demo_audit_logs() -> list[AuditLogEntry]:
    return [
        AuditLogEntry(
            id="audit_document_uploaded",
            action="document.upload_registered",
            actor="demo.admin@example.com",
            target="doc_upload_synthetic_vendor_policy_md",
            created_at=datetime(2026, 1, 15, 11, 50, tzinfo=UTC),
            metadata={"mode": "metadata_only", "status": "queued"},
        ),
        AuditLogEntry(
            id="audit_document_processed",
            action="document.processed",
            actor="system",
            target="doc_vendor_intake",
            created_at=datetime(2026, 1, 15, 12, 1, tzinfo=UTC),
            metadata={"chunk_count": "3", "next_step": "ready_for_rag_indexing"},
        ),
        AuditLogEntry(
            id="audit_rag_query_submitted",
            action="rag.query_submitted",
            actor="demo.admin@example.com",
            target="airun_what_risks_a",
            created_at=datetime(2026, 1, 15, 12, 5, tzinfo=UTC),
            metadata={"provider": "mock", "source_count": "3"},
        ),
        AuditLogEntry(
            id="audit_ai_run_created",
            action="ai_run.created",
            actor="demo.admin@example.com",
            target="airun_vendor_risk_summary",
            created_at=datetime(2026, 1, 15, 12, 10, tzinfo=UTC),
            metadata={"workflow": "Document Intake Review", "status": "needs_review"},
        ),
        AuditLogEntry(
            id="audit_review_created",
            action="review.created",
            actor="system",
            target="review_vendor_risk_summary",
            created_at=datetime(2026, 1, 15, 12, 12, tzinfo=UTC),
            metadata={"priority": "high", "source_document_id": "doc_vendor_intake"},
        ),
        AuditLogEntry(
            id="audit_review_approved",
            action="review.approved",
            actor="demo.reviewer@example.com",
            target="review_vendor_risk_summary",
            created_at=datetime(2026, 1, 15, 12, 18, tzinfo=UTC),
            metadata={"auditability": "human_decision", "decision": "approved"},
        ),
    ]
